src/client/src/containers/RightSidebar/index.tsx

```diff
@@ -123,17 +125,45 @@ class RightSidebar extends React.Component<Props, IRightSidebarState> {
   };
 
   public handleFrameworkChange = (option: IDropDownOptionType) => {
-    const { frontendFramework, pages } = this.props.selection;
-    const { vscode } = this.props;
+    const { frontendFramework, backendFramework, pages } = this.props.selection;
+    const {
+      vscode,
+      selectPages,
+      frontEndOptions,
+      selectFrontendFramework
+    } = this.props;
     if (frontendFramework.internalName !== option.value) {
       vscode.postMessage({
-        module: EXTENSION_MODULES.VSCODEUI,
-        command: EXTENSION_COMMANDS.RESET_PAGES,
-        track: false,
-        text: PAYLOAD_MESSAGES_TEXT.SWITCH_FRAMEWORKS_TEXT,
+        module: EXTENSION_MODULES.CORETS,
+        command: EXTENSION_COMMANDS.GET_PAGES,
         payload: {
-          internalName: option.value,
-          pagesLength: pages.length
+          projectType: WIZARD_CONTENT_INTERNAL_NAMES.FULL_STACK_APP,
+          frontendFramework: option.value,
+          backendFramework: backendFramework.internalName
+        }
+      });
+      const newPages: ISelected[] = pages.map(page => {
```

Comment:
not sure if i'm understanding but does this mean if you switch frameworks with [master detail, blank, list] in angular to react you will still have [master detail, blank, list]? If so great UX improvement, but wouldn't this also add in overhead to anyone who wants to contribute a page. Now one must know React, Angular and Vue before they can add a new page to webts. Or the pages must remain synchronized 100% of the time. 

Do you think that this is okay? Assuming that you intended that the team kept pages synchronized anyways, but feels like its harder for someone outside the team to contribute pages.

Reply:
Hmm, I see. We had this discussion with the team as well. But, we decided to go ahead with this because this would mean a huge improvement as far as user experience is concerned.
I agree that pages would need to be synchronized across all frameworks now. Yeahh, I am not very sure though.

Reply:
Yeah this is the main reason why we didn't do this, its a trade off either way. One thing that would be the best of both worlds is to check if it exists before adding it. But I don't know if that additional complexity is wanted?

Reply:
That could be a possibility. However, I would keep it as is in this PR. I shall create an issue for that to be done in the next Sprint though. Thanks for the insight - that seems good :)